### PR TITLE
feat(accountCreateTransaction): Implement `AccountCreateTransaction` E2E tests: TCK

### LINKED ISSUE #205: `TCK`: Implement JSON-RPC methods for `AccountCreateTransaction`
Implement the updated `AccountCreateTransaction` tests.

### PR DESCRIPTION
**Pull request by rwalworth**
This PR implements the tests documented in `test-specfications/crypto-service/accountCreateTransaction.md`. Since some of the work was already done for `AccountCreateTransaction` tests, this reworks those tests to match what is in the test specifications, in addition to adding any new tests. The docs were also updated to mark the tests as implemented.

Reminder, according to the development process we have established for the TCK (#202), this SHOULD NOT be approved by an SDK lead until their SDK has done any development required to run these tests against their SDK and those tests have passed. Feel free to provide comments and input to the PR, but approvals should not be granted until testing has occurred.

Issues opened in SDKs that encompass the SDK server work should be linked back to the TCK's `AccountCreateTransaction` development issue (#36) for easy tracking.

**Note**: This PR contains respective code changes for #209 and #211. If those change, this PR will change.

**Related issue(s)**:

Fixes #205 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by rwalworth** [comment]
Works with C++ SDK server (PR is [here](https://github.com/hashgraph/hedera-sdk-tck/pull/206)).

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 1]
```diff
@@ -293,13 +293,13 @@ https://docs.hedera.com/hedera/sdks-and-apis/rest-api
...
+| 5       | Creates an account with the staked account ID set to an empty account ID              | key=<VALID_KEY>, stakedAccountId=""                                                          | The account creation fails with and SDK internal error.                                               | Y                 |
+| 6       | Creates an account with the staked node ID set to an invalid node ID                  | key=<VALID_KEY>, stakedNodeId=-100                                                           | The account creation fails with an INVALID_STAKING_ID response code from the network.                 | Y                 |
+| 7       | Creates an account with a staked account ID and a staked node ID                      | key=<VALID_KEY>, stakedAccountId=<OPERATOR_ACCOUNT_ID>, stakedNodeId=<VALID_NETWORK_NODE_ID> | The account creation fails with an INVALID_STAKING_ID response code from the network.                 | Y                 |
```
Comment (rwalworth) [inline test-specifications/crypto-service/accountCreateTransaction.md:302]: Actually noticed an issue with this test (at least for C++ SDK). Since both staked account ID and node ID are being set, this test (which actually should pass) is dependent on the order that the protobuf fields are set in the SDK. For example, if the SDK when building the transactions or SDK server sets the staked node ID first and then the staked account ID, the created account is staked to the input staked account ID. However if the staked account ID is set first, the created account is staked to the input staked node ID. Is there/should there be an enforced order to these for the SDKs?

**Review by gsstoykov — COMMENTED** [review]
Regarding `test-specifications/crypto-service/` can we think of a better name than `crypto-service` here? From my perspective this is not that appropriate. Will it be better to have more specific test specification folders like `test-specifications/account`, `test-specifications/contract` and etc?

**Comment by rwalworth** [comment]
> Regarding `test-specifications/crypto-service/` can we think of a better name than `crypto-service` here? From my perspective this is not that appropriate. Will it be better to have more specific test specification folders like `test-specifications/account`, `test-specifications/contract` and etc?

Sure, the initial idea was to organize request tests into the Hedera service to which that request was handled (i.e. `AccountCreateTransaction` into `crypto-service`, `ContractCreateTransaction` would be in `smart-contract-service`, etc.), but I'm not attached to this and am fine with changing it. I can make a separate issue to change this

File: test/crypto-service/test_accountCreateTransaction.js [hunk 2]
```diff
@@ -0,0 +1,1005 @@
...
+        assert.equal(err.data.status, "AUTORENEW_DURATION_NOT_IN_RANGE");
+        return;
+      }
+      
+      // The test failed, no error was thrown.
+      assert.fail("Should throw an error");
+    });
+  });
+
+  describe("Memo", async function () {
```
Comment (0xivanov) [inline test/crypto-service/test_accountCreateTransaction.js:520]: We can add a  test case for asserting `INVALID_ZERO_BYTE_IN_STRING` error. Can be done by passing a string containing `\0`

File: test/crypto-service/test_accountCreateTransaction.js [hunk 3]
```diff
@@ -0,0 +1,1005 @@
...
+        const response = await JSONRPCRequest("createAccount", {
+          key: key.key,
+          maxAutoTokenAssociations: 5001,
+          commonTransactionParams: {
+            maxTransactionFee: 100000000000
+          }
+        });
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch (err) {
+        assert.equal(err.data.status, "REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT");
```
Comment (0xivanov) [inline test/crypto-service/test_accountCreateTransaction.js:693]: I think this status code was changed in https://github.com/hashgraph/hedera-services/issues/13687 . Now it returns `INVALID_MAX_AUTO_ASSOCIATIONS`

Comment (rwalworth) [reply test/crypto-service/test_accountCreateTransaction.js:716]: Yep, I can see this test failing now when testing with `--network-tag=0.52.0-alpha.5`. I'll fix this here when I can test it against C++ server

Comment (rwalworth) [reply test/crypto-service/test_accountCreateTransaction.js]: Updated to new response code and working with C++ server

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 4]
```diff
@@ -256,12 +257,12 @@ https://docs.hedera.com/hedera/sdks-and-apis/rest-api
...
-| 2       | Creates an account with a max token association that is the minimum value      | key=<VALID_KEY>, maxAutoTokenAssociations=0    | The account creation succeeds and the account has 0 automatic token associations.                                                | N                 |
-| 3       | Creates an account with a max token association that is the maximum value      | key=<VALID_KEY>, maxAutoTokenAssociations=1000 | The account creation succeeds and the account has 1000 automatic token associations.                                             | N                 |
-| 4       | Creates an account with a max token association that exceeds the maximum value | key=<VALID_KEY>, maxAutoTokenAssociations=1001 | The account creation fails with a REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT response code from the network. | N                 |
+| Test no | Name                                                                           | Input                                                                                                  | Expected response                                                                               | Implemented (Y/N) |
```
Comment (thenswan) [inline test-specifications/crypto-service/accountCreateTransaction.md:260]: Not sure when we should add case with `-1` value, but putting this comment as a reminder about such case.

Comment (jsync-swirlds) [reply test-specifications/crypto-service/accountCreateTransaction.md:260]: The `-1` case should generally be included for every signed numeric value.  The reason being that it's the longest encoded value in protobuf, and tests the binary maximum edge case.

File: test/crypto-service/test_accountCreateTransaction.js [hunk 5]
```diff
@@ -0,0 +1,1028 @@
+import { JSONRPCRequest } from "../../client.js";
+import mirrorNodeClient from "../../mirrorNodeClient.js";
+import consensusInfoClient from "../../consensusInfoClient.js";
+import { setOperator, getNodeType } from "../../setup_Tests.js";
```
Comment (0xivanov) [inline test/crypto-service/test_accountCreateTransaction.js:4]: `getNodeType` is unused

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 4 above)
Comment (thenswan) [reply test-specifications/crypto-service/accountCreateTransaction.md:260]: I mean in the context of HIP-904 (where the value `-1` is  permitted for `max_auto_associations`, and this value means “unlimited”).

Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md:260]: Created [this](https://github.com/hashgraph/hedera-sdk-tck/issues/229) to add them. Tests will need to be added here, `AccountUpdateTransaction`, and a couple contract transactions

File: test/crypto-service/test_accountCreateTransaction.js (hunk 5 above)
Comment (rwalworth) [reply test/crypto-service/test_accountCreateTransaction.js]: Removed

File: test-specifications/crypto-service/accountCreateTransaction.md (hunk 4 above)
Comment (thenswan) [reply test-specifications/crypto-service/accountCreateTransaction.md:260]: Thanks
cc @0xivanov

File: test/crypto-service/test_accountCreateTransaction.js (hunk 3 above)
Comment (0xivanov) [reply test/crypto-service/test_accountCreateTransaction.js:719]: Why did this get reverted?

Comment (rwalworth) [reply test/crypto-service/test_accountCreateTransaction.js]: This was a mistake, reverted back

**Review by 0xivanov — APPROVED** [review]
Approving since all the tests are passing in Go and Java

**Review by RickyLB — APPROVED** [review]
Tests work with Swift SDK
